Extension/src/Debugger/configurationProvider.ts

```diff
@@ -864,6 +886,118 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
             }
         }
     }
+
+    private async expand(config: vscode.DebugConfiguration, folder: vscode.WorkspaceFolder | undefined): Promise<void> {
+        const folderPath: string | undefined = folder?.uri.fsPath || vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+        const vars: ExpansionVars = config.variables ? config.variables : {};
+        vars.workspaceFolder = folderPath || '{workspaceFolder}';
+        vars.workspaceFolderBasename = folderPath ? path.basename(folderPath) : '{workspaceFolderBasename}';
+        const expansionOptions: ExpansionOptions = { vars, recursive: true };
+        return expandAllStrings(config, expansionOptions);
+    }
+
+    // Returns true when ALL steps succeed
+    private async deploySteps(config: vscode.DebugConfiguration, folder: vscode.WorkspaceFolder | undefined, cancellationToken?: vscode.CancellationToken): Promise<boolean> {
+        if (!config.deploySteps || config.deploySteps.length === 0) {
+            return true;
+        }
+
+        let succeeded: boolean = true;
+        const deployStart: number = new Date().getTime();
+
+        STEPS: for (const step of config.deploySteps) {
```

Comment:
Instead of using this strange syntax for breaking out of the `for` instead of the `switch`, can you move this whole for loop into a new function and have the `break STEPS;` become `return succeeded;` instead?  I believe that's the only piece of state that needs to be returned out of this block.